From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lukasz Sobala <lsobala1@gmail.com>
Date: Tue, 18 Aug 2026 21:38:48 +0200
Subject: arm64: dts: rockchip: rock-2f: Enable PCIe

Enable PCIe Gen2 x1 for the PCIe FPC connector. The combo PHY is
already enabled in rk3528-rock-2.dtsi and is free for PCIe because the
Type-C port is USB 2.0 only. Reset GPIO matches rk3528-rock-2f.dts in
the Radxa 6.1 kernel.

The connector only carries VCC5V0_SYS; the 3.3V for the card is made
on the attached board and switched by the connector's POWER-EN pin,
wired to GPIO1_A4 (PCIE_EN in the v1.01 schematic). Model it as the
enable GPIO of the vpcie3v3 regulator so the card actually gets power.

Signed-off-by: Lukasz Sobala <lsobala1@gmail.com>
---
diff --git a/arch/arm64/boot/dts/rockchip/rk3528-rock-2f.dts b/arch/arm64/boot/dts/rockchip/rk3528-rock-2f.dts
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/rockchip/rk3528-rock-2f.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3528-rock-2f.dts
@@ -7,6 +7,33 @@
 / {
 	model = "Radxa ROCK 2F";
 	compatible = "radxa,rock-2f", "rockchip,rk3528";
+
+	vcc3v3_pcie20: regulator-3v3-vcc-pcie20 {
+		compatible = "regulator-fixed";
+		enable-active-high;
+		gpios = <&gpio1 RK_PA4 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pcie_pwr_en>;
+		regulator-name = "vcc3v3_pcie20";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		startup-delay-us = <5000>;
+		vin-supply = <&vcc5v0_sys>;
+	};
+};
+
+&pcie {
+	reset-gpios = <&gpio1 RK_PA2 GPIO_ACTIVE_HIGH>;
+	vpcie3v3-supply = <&vcc3v3_pcie20>;
+	status = "okay";
+};
+
+&pinctrl {
+	pcie {
+		pcie_pwr_en: pcie-pwr-en {
+			rockchip,pins = <1 RK_PA4 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
 };

 &usb_host0_xhci {
--
Armbian

